# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/BaseTypes.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/compiler/translator/BaseTypes.h
--- a/gfx/angle/checkout/src/compiler/translator/BaseTypes.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/BaseTypes.h	Fri Nov 08 17:06:00 2019 +0000
@@ -622,6 +622,9 @@
 
     EvqDrawID,  // ANGLE_multi_draw
 
+    EvqBaseVertex,    // ANGLE_base_vertex_base_instance
+    EvqBaseInstance,  // ANGLE_base_vertex_base_instance
+
     // built-ins read by fragment shader
     EvqFragCoord,
     EvqFrontFacing,
@@ -690,6 +693,11 @@
     return (qualifier == EvqTemporary || qualifier == EvqGlobal);
 }
 
+inline bool IsStorageBuffer(TQualifier qualifier)
+{
+    return qualifier == EvqBuffer;
+}
+
 enum TLayoutImageInternalFormat
 {
     EiifUnspecified,
@@ -907,6 +915,8 @@
     case EvqPosition:               return "Position";
     case EvqPointSize:              return "PointSize";
     case EvqDrawID:                 return "DrawID";
+    case EvqBaseVertex:             return "BaseVertex";
+    case EvqBaseInstance:           return "BaseInstance";
     case EvqFragCoord:              return "FragCoord";
     case EvqFrontFacing:            return "FrontFacing";
     case EvqPointCoord:             return "PointCoord";